signing/signing.go

```diff
@@ -0,0 +1,50 @@
+package signing
+
+import (
+	"context"
+	"log"
+	"net/http"
+	"path/filepath"
+	"strings"
+
+	"github.com/gorilla/mux"
+	"github.com/ossf/scorecard/v2/cron/data"
+)
+
+type ScorecardOutput struct {
+	JsonOutput string
+}
+
+func GetResults(w http.ResponseWriter, r *http.Request) {
+	// Get params to build GCS filepath.
+	ctx := context.Background()
+	bucketURL := "gs://ossf-scorecard-results"
+	host := mux.Vars(r)["host"]
+	orgName := mux.Vars(r)["orgName"]
+	repoName := mux.Vars(r)["repoName"]
+	resultsFile := filepath.Join(host, orgName, repoName, "results.json")
+
+	// Sanitize input and log query.
+	resultsFile = filepath.Clean(resultsFile)
+	matched, err := filepath.Match("*/*/*/results.json", resultsFile)
+	if err != nil || !matched {
+		http.Error(w, "error verifying filepath format", http.StatusInternalServerError)
+		log.Println("matched filepath?", matched, err)
+		return
+	}
+	resultsFileEscaped := strings.Replace(resultsFile, "\n", "", -1)
```

Comment:
CodeQL wanted stronger cleaning: https://github.com/ossf/scorecard-webapp/security/code-scanning/43

But I think filepath.Clean should be enough because it gets rid of all the escaping. Should I just add a nolint to this line?

Reply:
do you know why there are `\n` and `\r` in the path? Was it because we don't trust the input?

Reply:
ok got it.. sorry I've been slow understanding this :) I suppose it's fine to remove these, just add a comment.
Shall we also truncate the string before logging? Maybe 256 characters max? (We can use the maximum Linux path length, I think). If the string is greater, we can return an error

Reply:
No problem, I added a check for this